### DISCUSSION TITLE
Suggestion queries don't work on some nodes

### DISCUSSION THREAD

**Discussion by Vall3n** [ravendb/ravendb#18875] [Q&A]
The client app received either zero suggestions or the expected results intermittently when executing suggestion queries.

Trying to replicate using a simple suggestion query in Studio I noticed that Node A returned results consistently, but Nodes B & C always returned 0 suggestions.

```
from index 'Locations/Autocomplete'
select suggest(NameOrTag, "fo0*")
```

I have a cluster with 3 nodes hosted on https://cloud.ravendb.net/ running on 5.4 LTS.
All 3 nodes appeared healthy according to the Database Group Topology.

Any idea why the 2 nodes would return no results, or how I can identify and fix this in the future?

**Comment by ayende** [comment]
That shouldn't generally be the case.
They are running the same logic on both systems.
Did you disable indexing on some of the nodes?

Can you create a support ticket on the cloud about this?

**Comment by Vall3n** [comment]
No, this was recently set up as a new cluster, with all settings left to defaults.
I did not disable indexes or do anything different on the bad nodes, in fact I never even logged into them directly (Studio) until I needed to investigate further why the client sometimes received 0 suggestions for the exact same query during testing.

The data was migrated from a v3 instance, and the indexes were recreated from code (on app startup, code modified to be v5 compatible) and then automatically synced across the nodes.
Normal document queries were working, only Suggestion queries returned 0 results on the bad nodes.

As a quick "fix" I deleted the bad nodes and recreated them. After they finished automatic syncing from the healthy node A, all appeared to work.

The question here remains how to debug this if it happens in the future? 
Are there any other options other than recreating nodes?

With the risk of:
- What if all nodes go bad?
- Is hard to detect if it stops working for only some users/sessions (if a single node goes bad), are there any other signs to look out for?
   - It doesn't seem right to code around the issue and flag 0 results as warnings, as that would essentially be checking if the db works as expected; and 0 results are often correct sometimes


I won't open a support ticket right now as the broken nodes have been permanently deleted.
I will keep an eye on the cluster for a while, hopefully it was a once-off initial migration issue..

**Reply by ayende** [reply]
See this post for details: 
https://ayende.com/blog/201473-A/with-bugs-failures-and-errors-ever-chugging-forward?key=0f0444b820284fccb5e51d70451d72eb

In short, that isn't user visible, since the cluster will redirect you around that.
